types/xhr-mock/index.d.ts

```diff
@@ -0,0 +1,57 @@
+// Type definitions for xhr-mock 1.9
+// Project: https://github.com/jameslnewell/xhr-mock#readme
+// Definitions by: Joscha Feth <https://github.com/joscha>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace mock {
+    interface Headers {
+        [k: string]: string;
+    }
+
+    interface MockResponse {
+        status(code: number): this;
+        status(): number;
+        statusText(statusText: string): this;
+        statusText(): string;
+        header(name: string, value: string): this;
+        header(name: string): string | null;
+        headers(obj: Headers): this;
+        headers(): Headers;
+        body(body: string): this;
+        body(): string;
+        timeout(timeout: boolean | number): this;
+        timeout(): boolean | number;
+    }
+
+    interface MockRequest {
+        method(): string;
+        url(): string;
+        query(): string;
+        header(name: string, value: string): this;
+        header(name: string): string | null;
+        headers(obj: Headers): this;
+        headers(): Headers;
+        body(body: string): this;
+        body(): string;
+        progress(loaded: number, total: number, lengthComputable?: boolean): void;
+    }
+
+    type MockFunction = (req: MockRequest, res: MockResponse) => MockResponse | null;
+
+    interface XhrMock {
+        XMLHttpRequest: XMLHttpRequest;
+        setup(): this;
+        teardown(): this;
+        reset(): this;
+        mock(method: string, url: string, fn: mock.MockFunction): this;
+        get(url: string, fn: mock.MockFunction): this;
+        post(url: string, fn: mock.MockFunction): this;
+        put(url: string, fn: mock.MockFunction): this;
+        patch(url: string, fn: mock.MockFunction): this;
+        delete(url: string, fn: mock.MockFunction): this;
+    }
+}
+
+declare var mock: mock.XhrMock;
+export = mock;
+export as namespace mock;
```

Comment:
This is to enable usage as a UMD module (i.e. a global). This is also what your tests use. But is this actually allowed by the library? The documentation only shows `require()` usage.

Reply:
@minestarks - it was the only way I could make the synthetic interfaces (request, response) available in the target code. I wasn't aware this also meant exposure as a global, so I think you are right and it is incorrect. Do you have a suggestion about how to make the interfaces available instead?

Reply:
Once you've imported the module via `import mock = require('xhr-mock');`, you should be able to reference your interfaces from user code like `mock.MockResponse` and `mock.MockResponse`.

Reply:
fixed in c88276a0d